packages/docusaurus-github-stars/src/utils/constants.ts

```diff
@@ -1,7 +1,10 @@
+import { packageUpSync } from 'package-up';
 import fs from 'node:fs';
 
 function getPluginName() {
-  const packageJson = JSON.parse(fs.readFileSync('package.json', 'utf-8'));
+  const packageStr = packageUpSync({ cwd: __dirname });
+  if (!packageStr) throw new Error('Could not find package.json');
+  const packageJson = JSON.parse(fs.readFileSync(packageStr, 'utf-8')) as { name: string };
```

Comment:
I could have sworn that Typescript had a thing for importing JSON files (https://www.typescriptlang.org/tsconfig/#resolveJsonModule ?), can we use that?

Reply:
Good call, a standard `require` call works. I also moved these utils into a reusable repo package.